internal/selectors/selectors.go

```diff
@@ -158,19 +158,75 @@ func GetNoneSelector(inst *unstructured.Unstructured) (bool, error) {
 	return noneSelector, nil
 }
 
-// cmExclusions are known (istio and kube-root) CA configmap which are excluded from propagation
-var cmExclusions = []string{"istio-ca-root-cert", "kube-root-ca.crt"}
+type ExclusionByNameSpec struct {
+	// Group of the resource defined below. This is used to unambiguously identify the resource. It
+	// may be omitted for core resources (e.g. "secrets").
+	Group string
+
+	// Kind of the resource
+	Kind string
+
+	// Name of the resoruce - can be used to filter by name of the object
+	Name string
+}
+
+type ExclusionByLabelsSpec struct {
+	// Group of the resource defined below. This is used to unambiguously identify the resource. It
+	// may be omitted for core resources (e.g. "secrets").
+	Group string
+
+	// Kind of the resource
+	Kind string
+
+	// Labels - can be used to filter by label; both key and value are used to match the label, i.e.
+	// value of the label has also to match. If multiple labels are set, ALL of them have to match!
+	Labels map[string]string
+}
+
+// ExclusionsByName are known (istio and kube-root) CA configmap which are excluded from propagation
+var ExclusionsByName = []ExclusionByNameSpec{
+	{Kind: "ConfigMap", Name: "istio-ca-root-cert"},
+	{Kind: "ConfigMap", Name: "kube-root-ca.crt"},
+}
+
+var ExclusionByLabels = []ExclusionByLabelsSpec{
+	{Kind: "ServiceAccount", Labels: map[string]string{"cattle.io/creator": "norman"}},
+	{Group: "rbac.authorization.k8s.io", Kind: "RoleBinding", Labels: map[string]string{"cattle.io/creator": "norman"}},
+	{Group: "rbac.authorization.k8s.io", Kind: "Role", Labels: map[string]string{"cattle.io/creator": "norman"}},
+	{Group: "networking.k8s.io", Kind: "NetworkPolicy", Labels: map[string]string{"cattle.io/creator": "norman"}},
+}
 
 // isExcluded returns true to indicate that this object is excluded from being propagated
 func isExcluded(inst *unstructured.Unstructured) (bool, error) {
 	name := inst.GetName()
 	kind := inst.GetKind()
 	group := inst.GroupVersionKind().Group
+	labels := inst.GetLabels()
 
-	for _, excludedResourceName := range cmExclusions {
-		if group == "" && kind == "ConfigMap" && name == excludedResourceName {
+	// exclusion by name
+	for _, res := range ExclusionsByName {
+		if group == res.Group && kind == res.Kind && name == res.Name {
 			return true, nil
 		}
 	}
+
+	// exclusion by labels
+	for _, res := range ExclusionByLabels {
+		// check group and kind first
+		if group == res.Group && kind == res.Kind {
+			// check all specified labels
+			allLabelsFound := true
+			for excludedKey, excludedValue := range res.Labels {
+				labelValue, ok := labels[excludedKey]
+				if !ok || excludedValue != labelValue {
+					allLabelsFound = false
+				}
+			}
+			if allLabelsFound {
+				return true, nil
+			}
```

Comment:
Does this need to be so complex? Right now, all of the proposed exclusions only look at a single label. If all known cases would be fine with a single key-value pair, let's simplify it to that and only add this complexity when we need it?